app/jobs/weekly_email.rb

```diff
@@ -0,0 +1,11 @@
+class WeeklyEmail < ApplicationJob
+  #sidekiq_options retry: false
+   queue_as :default
+   def perform(*)
+    @users = User.all
+    users.each do |user|
+      mail = ApplicationMailer.sample_email(id: user.id, email: user.email)
```

Comment:
You need to pass the `user` variable, check the declaration of `sample_email`, it is accepting only one parameter and you are sending two.